core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -95,7 +95,9 @@ class ReplicaFetcherThread(name: String,
   private val minBytes = brokerConfig.replicaFetchMinBytes
   private val maxBytes = brokerConfig.replicaFetchResponseMaxBytes
   private val fetchSize = brokerConfig.replicaFetchMaxBytes
-  private val shouldSendLeaderEpochRequest: Boolean = brokerConfig.interBrokerProtocolVersion >= KAFKA_0_11_0_IV2
+  private val brokerSupportsLeaderEpochRequest: Boolean = brokerConfig.interBrokerProtocolVersion >= KAFKA_0_11_0_IV2 &&
+    brokerConfig.logMessageFormatVersion >= KAFKA_0_11_0_IV2
```

Comment:
It is possible for the default message format version to be lower than what is configured for the topic, so I'm not sure this addition makes sense.